lib/build-code.js

```diff
@@ -0,0 +1,82 @@
+
+import path from 'path';
+import _fs from 'fs';
+import _ from 'lodash';
+import B from 'bluebird';
+
+let fs = {
+  readFile: B.promisify(_fs.readFile)
+};
+
+let root = path.resolve(__dirname, '../uiauto');
+
+// keep a list of files we examine for the purposes of making sure we're
+// not adding the same file to the collated version twice
+let filesExamined;
+
+// this regex helps us get the file path of an import
+let importRe = /^#import ('|")([^('|")]+)('|")$/mg;
+
+async function getDepsForFile (file, extraImports = []) {
+  if (_.contains(filesExamined, file)) {
+    throw new Error(`Re-examining file ${file}; you need to make sure ` +
+                    `the graph is set up so we don't require files twice`);
+  }
+  filesExamined.push(file);
+  
+  let data = await fs.readFile(file, 'utf8');
+  let deps = {
+    [file]: []
+  };
+
+  let matches = [];
+  let match = importRe.exec(data);
+  while (match) {
+    if (match) {
+      matches.push(match[2]);
+    }
+    match = importRe.exec(data);
+  }
+  if (extraImports && extraImports.length > 0) {
```

Comment:
`extraImports` will always exist, and there is no need to check for length as `concat` works reasonably with an empty array. So no need for the conditional.